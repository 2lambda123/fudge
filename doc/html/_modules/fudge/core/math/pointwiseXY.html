
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>fudge.core.math.pointwiseXY &mdash; Fudge and GND 1.2 beta documentation</title>
    
    <link rel="stylesheet" href="../../../../_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../../',
        VERSION:     '1.2 beta',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="../../../../_static/gnd-20121206-favicon.ico"/>
    <link rel="top" title="Fudge and GND 1.2 beta documentation" href="../../../../index.html" />
    <link rel="up" title="Module code" href="../../../index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../../../../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../../../../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li><a href="../../../../index.html">Fudge and GND 1.2 beta documentation</a> &raquo;</li>
          <li><a href="../../../index.html" accesskey="U">Module code</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <h1>Source code for fudge.core.math.pointwiseXY</h1><div class="highlight"><pre>
<span class="c"># &lt;&lt;BEGIN-copyright&gt;&gt;</span>
<span class="c"># &lt;&lt;END-copyright&gt;&gt;</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">This module defines &#39;pointwiseXY&#39; in pure python. Users will generally prefer to use &#39;pointwiseXY_C&#39;,</span>
<span class="sd">which is a faster and more capable C implementation.</span>
<span class="sd">If the C version is not available, fudge will revert to this version.</span>

<span class="sd">cmattoon, July 2011</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="kn">from</span> <span class="nn">endl2dmathClasses</span> <span class="kn">import</span> <span class="n">endl2dmath</span>

<span class="n">__metaclass__</span> <span class="o">=</span> <span class="nb">type</span>

<div class="viewcode-block" id="pointwiseXY"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY">[docs]</a><span class="k">class</span> <span class="nc">pointwiseXY</span><span class="p">(</span> <span class="n">endl2dmath</span> <span class="p">):</span>
    <span class="k">def</span> <span class="nf">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">initialSize</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">overflowSize</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="bp">None</span><span class="p">,</span> <span class="n">dataForm</span><span class="o">=</span><span class="s">&quot;xys&quot;</span><span class="p">,</span> <span class="n">accuracy</span><span class="o">=</span><span class="bp">None</span><span class="p">,</span> 
            <span class="n">biSectionMax</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">interpolation</span><span class="o">=</span><span class="bp">None</span><span class="p">,</span> <span class="n">infill</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">safeDivide</span><span class="o">=</span><span class="bp">False</span><span class="p">):</span>

        <span class="k">if</span><span class="p">(</span> <span class="n">dataForm</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span> <span class="o">==</span> <span class="s">&#39;xsandys&#39;</span> <span class="p">)</span> <span class="p">:</span>
            <span class="n">data</span> <span class="o">=</span> <span class="nb">zip</span><span class="p">(</span><span class="o">*</span><span class="n">data</span><span class="p">)</span>
        <span class="k">if</span><span class="p">(</span> <span class="nb">type</span><span class="p">(</span> <span class="n">interpolation</span> <span class="p">)</span> <span class="ow">is</span> <span class="nb">str</span> <span class="p">)</span> <span class="p">:</span>
            <span class="k">if</span><span class="p">(</span> <span class="s">&#39;flat&#39;</span> <span class="ow">in</span> <span class="n">interpolation</span> <span class="p">)</span> <span class="p">:</span> <span class="n">interpolation</span> <span class="o">=</span> <span class="s">&#39;linear,linear&#39;</span>     <span class="c"># This is a kludge</span>
            <span class="n">interpolation</span> <span class="o">=</span> <span class="p">{</span> <span class="s">&#39;linear,linear&#39;</span> <span class="p">:</span> <span class="mi">0</span><span class="p">,</span> <span class="s">&#39;log,linear&#39;</span> <span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;linear,log&#39;</span> <span class="p">:</span> <span class="mi">2</span><span class="p">,</span> <span class="s">&#39;log,log&#39;</span> <span class="p">:</span> <span class="mi">3</span> <span class="p">}[</span><span class="n">interpolation</span><span class="p">]</span>
            <span class="n">endl2dmath</span><span class="o">.</span><span class="n">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span> <span class="n">interpolation</span><span class="o">=</span><span class="n">interpolation</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">endl2dmath</span><span class="o">.</span><span class="n">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">)</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">accuracy</span> <span class="o">=</span> <span class="n">accuracy</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">biSectionMax</span> <span class="o">=</span> <span class="n">biSectionMax</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">infill</span> <span class="o">=</span> <span class="n">infill</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">safeDivide</span> <span class="o">=</span> <span class="n">safeDivide</span>

    <span class="c"># override some endl2dmath functions for compatibility with pointwiseXY_C:</span>
    <span class="k">def</span> <span class="nf">__add__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">):</span>
        <span class="k">return</span> <span class="n">pointwiseXY</span><span class="p">(</span> <span class="n">data</span> <span class="o">=</span> <span class="n">endl2dmath</span><span class="o">.</span><span class="n">__add__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span><span class="n">other</span><span class="p">)</span> <span class="p">)</span>

    <span class="k">def</span> <span class="nf">__mul__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">):</span>
        <span class="k">return</span> <span class="n">pointwiseXY</span><span class="p">(</span> <span class="n">data</span> <span class="o">=</span> <span class="n">endl2dmath</span><span class="o">.</span><span class="n">__mul__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span><span class="n">other</span><span class="p">)</span> <span class="p">)</span>

    <span class="k">def</span> <span class="nf">__getitem__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">index</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span><span class="p">[</span><span class="n">index</span><span class="p">]</span>

    <span class="k">def</span> <span class="nf">__setitem__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">index</span><span class="p">,</span> <span class="n">value</span><span class="p">):</span>
        <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">value</span><span class="p">)</span> <span class="ow">in</span> <span class="p">(</span><span class="nb">int</span><span class="p">,</span><span class="nb">float</span><span class="p">):</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">data</span><span class="p">[</span><span class="n">index</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="n">value</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">data</span><span class="p">[</span><span class="n">index</span><span class="p">]</span> <span class="o">=</span> <span class="n">value</span>

<div class="viewcode-block" id="pointwiseXY.getAccuracy"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.getAccuracy">[docs]</a>    <span class="k">def</span> <span class="nf">getAccuracy</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">accuracy</span>
</div>
<div class="viewcode-block" id="pointwiseXY.getBiSectionMax"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.getBiSectionMax">[docs]</a>    <span class="k">def</span> <span class="nf">getBiSectionMax</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">biSectionMax</span>
</div>
<div class="viewcode-block" id="pointwiseXY.getInfill"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.getInfill">[docs]</a>    <span class="k">def</span> <span class="nf">getInfill</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">infill</span>
</div>
<div class="viewcode-block" id="pointwiseXY.getSafeDivide"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.getSafeDivide">[docs]</a>    <span class="k">def</span> <span class="nf">getSafeDivide</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">safeDivide</span>
</div>
<div class="viewcode-block" id="pointwiseXY.thin"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.thin">[docs]</a>    <span class="k">def</span> <span class="nf">thin</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">accuracy</span><span class="p">):</span>
        <span class="k">return</span> <span class="n">pointwiseXY</span><span class="p">(</span> <span class="n">data</span> <span class="o">=</span> <span class="n">endl2dmath</span><span class="o">.</span><span class="n">thin</span><span class="p">(</span> <span class="bp">self</span><span class="p">,</span> <span class="n">accuracy</span> <span class="p">)</span> <span class="p">)</span>
</div>
<div class="viewcode-block" id="pointwiseXY.union"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.union">[docs]</a>    <span class="k">def</span> <span class="nf">union</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">,</span> <span class="n">fillWithSelf</span><span class="p">,</span> <span class="n">trim</span><span class="p">):</span>
        <span class="n">d</span> <span class="o">=</span> <span class="n">endl2dmath</span><span class="o">.</span><span class="n">union</span><span class="p">(</span> <span class="bp">self</span><span class="p">,</span> <span class="n">other</span> <span class="p">)</span>
        <span class="k">return</span> <span class="n">pointwiseXY</span><span class="p">(</span> <span class="n">data</span><span class="o">=</span><span class="n">d</span><span class="o">.</span><span class="n">data</span><span class="p">,</span> <span class="n">interpolation</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">interpolation</span> <span class="p">)</span>
</div>
<div class="viewcode-block" id="pointwiseXY.xSlice"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.xSlice">[docs]</a>    <span class="k">def</span> <span class="nf">xSlice</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">xMin</span><span class="p">,</span> <span class="n">xMax</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="bp">None</span><span class="p">,</span> <span class="n">dullEps</span><span class="o">=</span><span class="bp">None</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span>
        <span class="n">d</span> <span class="o">=</span> <span class="n">endl2dmath</span><span class="o">.</span><span class="n">slicex</span><span class="p">(</span> <span class="bp">self</span><span class="p">,</span> <span class="n">xMin</span><span class="p">,</span> <span class="n">xMax</span> <span class="p">)</span>
        <span class="k">return</span> <span class="n">pointwiseXY</span><span class="p">(</span> <span class="n">data</span><span class="o">=</span><span class="n">d</span><span class="o">.</span><span class="n">data</span><span class="p">,</span> <span class="n">interpolation</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">interpolation</span> <span class="p">)</span>
</div>
<div class="viewcode-block" id="pointwiseXY.copyDataToXYs"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.copyDataToXYs">[docs]</a>    <span class="k">def</span> <span class="nf">copyDataToXYs</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">xScale</span><span class="o">=</span><span class="bp">None</span><span class="p">,</span> <span class="n">yScale</span><span class="o">=</span><span class="bp">None</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">xScale</span> <span class="ow">or</span> <span class="n">yScale</span><span class="p">:</span>
            <span class="n">xScale</span> <span class="o">=</span> <span class="n">xScale</span> <span class="ow">or</span> <span class="mi">1</span><span class="p">;</span> <span class="n">yScale</span> <span class="o">=</span> <span class="n">yScale</span> <span class="ow">or</span> <span class="mi">1</span>
            <span class="k">return</span> <span class="p">[[</span><span class="n">xScale</span><span class="o">*</span><span class="n">x</span><span class="p">,</span><span class="n">yScale</span><span class="o">*</span><span class="n">y</span><span class="p">]</span> <span class="k">for</span> <span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">)</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span><span class="p">]</span>
        <span class="kn">import</span> <span class="nn">copy</span>
        <span class="k">return</span> <span class="n">copy</span><span class="o">.</span><span class="n">deepcopy</span><span class="p">(</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="p">)</span>
</div>
<div class="viewcode-block" id="pointwiseXY.copy"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.copy">[docs]</a>    <span class="k">def</span> <span class="nf">copy</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">return</span> <span class="n">pointwiseXY</span><span class="p">(</span> <span class="n">data</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">data</span><span class="p">,</span> <span class="n">accuracy</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">accuracy</span><span class="p">,</span> <span class="n">biSectionMax</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">biSectionMax</span><span class="p">,</span> 
                <span class="n">interpolation</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">interpolation</span><span class="p">,</span> <span class="n">infill</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">infill</span><span class="p">,</span> <span class="n">safeDivide</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">safeDivide</span> <span class="p">)</span>
</div>
<div class="viewcode-block" id="pointwiseXY.changeInterpolation"><a class="viewcode-back" href="../../../../fudge/fudge.core.math.html#fudge.core.math.pointwiseXY.pointwiseXY.changeInterpolation">[docs]</a>    <span class="k">def</span> <span class="nf">changeInterpolation</span><span class="p">(</span> <span class="bp">self</span><span class="p">,</span> <span class="n">interpolation</span><span class="p">,</span> <span class="n">accuracy</span> <span class="o">=</span> <span class="bp">None</span><span class="p">,</span> <span class="n">lowerEps</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span> <span class="n">upperEps</span> <span class="o">=</span> <span class="mi">0</span> <span class="p">)</span> <span class="p">:</span>
        <span class="k">if</span> <span class="n">interpolation</span> <span class="o">!=</span> <span class="s">&quot;linear,linear&quot;</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">Exception</span><span class="p">(</span><span class="s">&quot;Only linear interpolation currently supported&quot;</span><span class="p">)</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">toInterpolation</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="n">accuracy</span><span class="p">)</span>
</div>
    <span class="k">def</span> <span class="nf">__str__</span><span class="p">(</span> <span class="bp">self</span> <span class="p">)</span> <span class="p">:</span>

        <span class="k">return</span><span class="p">(</span> <span class="n">endl2dmath</span><span class="o">.</span><span class="n">toString</span><span class="p">(</span> <span class="bp">self</span> <span class="p">)</span> <span class="p">)</span></div>
</pre></div>

          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../../../../index.html">
              <img class="logo" src="../../../../_static/gnd.png" alt="Logo"/>
            </a></p>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../../../../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../../../../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../../../../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li><a href="../../../../index.html">Fudge and GND 1.2 beta documentation</a> &raquo;</li>
          <li><a href="../../../index.html" >Module code</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2012, B.Beck, D. Brown, C. Mattoon, N. Patel, N. Summers.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>