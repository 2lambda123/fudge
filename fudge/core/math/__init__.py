# <<BEGIN-copyright>>
# <<END-copyright>>

from math import *
import fudgemath
import fudge2dGrouping
import endl1dmathClasses
import endl2dmathClasses
import endl3dmathClasses
import endl4dmathClasses
