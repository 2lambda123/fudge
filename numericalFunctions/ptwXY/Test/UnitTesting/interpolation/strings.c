/*
# <<BEGIN-copyright>>
# Copyright (c) 2011, Lawrence Livermore National Security, LLC.
# Produced at the Lawrence Livermore National Laboratory.
# Written by the LLNL Computational Nuclear Physics group
#         (email: mattoon1@llnl.gov)
# LLNL-CODE-494171 All rights reserved.
# 
# This file is part of the FUDGE package (For Updating Data and 
#         Generating Evaluations)
# 
# When citing FUDGE, please use the following reference:
#   C.M. Mattoon, B.R. Beck, N.R. Patel, N.C. Summers, G.W. Hedstrom, D.A. Brown, "Generalized Nuclear Data: A New Structure (with Supporting Infrastructure) for Handling Nuclear Data", Nuclear Data Sheets, Volume 113, Issue 12, December 2012, Pages 3145-3171, ISSN 0090-3752, http://dx.doi.org/10. 1016/j.nds.2012.11.008
# 
# 
#     Please also read this link - Our Notice and Modified BSD License
# 
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions are met:
#     * Redistributions of source code must retain the above copyright
#       notice, this list of conditions and the following disclaimer.
#     * Redistributions in binary form must reproduce the above copyright
#       notice, this list of conditions and the following disclaimer in the
#       documentation and/or other materials provided with the distribution.
#     * Neither the name of Lawrence Livermore National Security, LLC. nor the
#       names of its contributors may be used to endorse or promote products
#       derived from this software without specific prior written permission.
# 
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
# DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY BE LIABLE FOR ANY
# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
# <<END-copyright>>
*/

#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <stdarg.h>

#include <ptwXY.h>
#include <ptwXY_utilities.h>

static int verbose = 0;

nfu_status chargedParticleGetValue( void *argList, double x, double *y, double x1, double y1, double x2, double y2 );
void printMsg( const char *fmt, ... );
/*
****************************************************************
*/
int main( int argc, char **argv ) {

    int iarg, echo = 0, nPoints;
    ptwXYPoints *ptwXY, *ptwXY2, *ptwXY3;
    double accuracy = 1e-3;
    double xys[] = { 1, 1, 10, 10 };
    nfu_status status = nfu_Okay;

    for( iarg = 1; iarg < argc; iarg++ ) {
        if( strcmp( "-e", argv[iarg] ) == 0 ) {
            echo = 1; }
        else if( strcmp( "-v", argv[iarg] ) == 0 ) {
            verbose = 1; }
        else {
            printMsg( "Error %s: invalid input option '%s'", __FILE__, argv[iarg] );
        }
    }
    if( echo ) fprintf( stderr, "%s\n", __FILE__ );
    
    nPoints = sizeof( xys ) / sizeof( xys[0] ) / 2;

    if( ( ptwXY2 = ptwXY_create( ptwXY_interpolationOther, "charged-particle", 5, accuracy, 10, 10, nPoints, xys, &status, 0 ) ) == NULL ) 
        printMsg( "ptwXY2 creation: status = %d: %s", status, nfu_statusMessage( status ) );

    if( ( ptwXY = ptwXY_create( ptwXY_interpolationLinLin, NULL, 5, accuracy, 10, 10, nPoints, xys, &status, 0 ) ) == NULL ) 
        printMsg( "ptwXY creation: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "linear-linear string = <%s>\n", ptwXY_getInterpolationString( ptwXY ) );
    if( ptwXY_copy( ptwXY2, ptwXY ) != nfu_Okay ) printMsg( "ptwXY_copy: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY2 ) );
    if( ( ptwXY3 = ptwXY_clone( ptwXY2, &status ) ) == NULL ) printMsg( "ptwXY_clone: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY3 ) );
    ptwXY_free( ptwXY );
    ptwXY_free( ptwXY3 );

    if( verbose ) printf( "\n" );
    if( ( ptwXY = ptwXY_create( ptwXY_interpolationLinLog, NULL, 5, accuracy, 10, 10, nPoints, xys, &status, 0 ) ) == NULL ) 
        printMsg( "ptwXY creation: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "linear-log string = <%s>\n", ptwXY_getInterpolationString( ptwXY ) );
    if( ptwXY_copy( ptwXY2, ptwXY ) != nfu_Okay ) printMsg( "ptwXY_copy: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY2 ) );
    if( ( ptwXY3 = ptwXY_clone( ptwXY2, &status ) ) == NULL ) printMsg( "ptwXY_clone: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY3 ) );
    ptwXY_free( ptwXY );
    ptwXY_free( ptwXY3 );

    if( verbose ) printf( "\n" );
    if( ( ptwXY = ptwXY_create( ptwXY_interpolationLogLin, NULL, 5, accuracy, 10, 10, nPoints, xys, &status, 0 ) ) == NULL ) 
        printMsg( "ptwXY creation: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "log-linear string = <%s>\n", ptwXY_getInterpolationString( ptwXY ) );
    if( ptwXY_copy( ptwXY2, ptwXY ) != nfu_Okay ) printMsg( "ptwXY_copy: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY2 ) );
    if( ( ptwXY3 = ptwXY_clone( ptwXY2, &status ) ) == NULL ) printMsg( "ptwXY_clone: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY3 ) );
    ptwXY_free( ptwXY );
    ptwXY_free( ptwXY3 );

    if( verbose ) printf( "\n" );
    if( ( ptwXY = ptwXY_create( ptwXY_interpolationLogLog, NULL, 5, accuracy, 10, 10, nPoints, xys, &status, 0 ) ) == NULL ) 
        printMsg( "ptwXY creation: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "log-log string = <%s>\n", ptwXY_getInterpolationString( ptwXY ) );
    if( ptwXY_copy( ptwXY2, ptwXY ) != nfu_Okay ) printMsg( "ptwXY_copy: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY2 ) );
    if( ( ptwXY3 = ptwXY_clone( ptwXY2, &status ) ) == NULL ) printMsg( "ptwXY_clone: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY3 ) );
    ptwXY_free( ptwXY );
    ptwXY_free( ptwXY3 );

    if( verbose ) printf( "\n" );
    if( ( ptwXY = ptwXY_create( ptwXY_interpolationFlat, NULL, 5, accuracy, 10, 10, nPoints, xys, &status, 0 ) ) == NULL ) 
        printMsg( "ptwXY creation: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "flat string = <%s>\n", ptwXY_getInterpolationString( ptwXY ) );
    if( ptwXY_copy( ptwXY2, ptwXY ) != nfu_Okay ) printMsg( "ptwXY_copy: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY2 ) );
    if( ( ptwXY3 = ptwXY_clone( ptwXY2, &status ) ) == NULL ) printMsg( "ptwXY_clone: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY3 ) );
    ptwXY_free( ptwXY );
    ptwXY_free( ptwXY3 );

    if( verbose ) printf( "\n" );
    if( ( ptwXY = ptwXY_create( ptwXY_interpolationOther, "charged-particle", 5, accuracy, 10, 10, nPoints, xys, &status, 0 ) ) == NULL ) 
        printMsg( "ptwXY creation: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "other string = <%s>\n", ptwXY_getInterpolationString( ptwXY ) );
    if( ptwXY_copy( ptwXY2, ptwXY ) != nfu_Okay ) printMsg( "ptwXY_copy: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY2 ) );
    if( ( ptwXY3 = ptwXY_clone( ptwXY2, &status ) ) == NULL ) printMsg( "ptwXY_clone: status = %d: %s", status, nfu_statusMessage( status ) );
    if( verbose ) printf( "string = <%s>\n", ptwXY_getInterpolationString( ptwXY3 ) );
    ptwXY_free( ptwXY );
    ptwXY_free( ptwXY3 );

    ptwXY_free( ptwXY2 );

    exit( EXIT_SUCCESS );
}
/*
****************************************************************
*/
nfu_status chargedParticleGetValue( void *argList, double x, double *y, double x1, double y1, double x2, double y2 ) {

    double A, B, T = *((double *) argList);

    B = log( x2 * y2 / ( x1 * y1 ) ) / ( 1. / sqrt( x1 - T ) - 1. / sqrt( x2 - T ) );
    A = x1 * y1 * exp( B / sqrt( x1 - T ) );
    *y = A * exp( - B / sqrt( x - T ) ) / x;
    return( nfu_Okay );
}
/*
****************************************************************
*/
void printMsg( const char *fmt, ... ) {

    va_list args;

    va_start( args, fmt );
    vfprintf( stderr, fmt, args );
    fprintf( stderr, "\n" );
    va_end( args );
    exit( EXIT_FAILURE );
}
