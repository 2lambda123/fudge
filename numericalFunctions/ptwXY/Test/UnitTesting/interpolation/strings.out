linear-linear string = <lin,lin>
string = <lin,lin>
string = <lin,lin>

linear-log string = <lin,log>
string = <lin,log>
string = <lin,log>

log-linear string = <log,lin>
string = <log,lin>
string = <log,lin>

log-log string = <log,log>
string = <log,log>
string = <log,log>

flat string = <flat>
string = <flat>
string = <flat>

other string = <charged-particle>
string = <charged-particle>
string = <charged-particle>
